cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1424,9 +1434,19 @@ public Map<ProgramRunId, RunRecordDetail> getActiveRuns(ApplicationId applicatio
    * @param programId given program
    * @return map of run id to run record meta
    */
-  public Map<ProgramRunId, RunRecordDetail> getActiveRuns(ProgramId programId) throws IOException {
+  public Map<ProgramRunId, RunRecordDetail> getActiveRuns(ProgramId programId)
```

Comment:
So implicitly we're changing the version inside `ProgramRunId`, we're somehow changing the contract here... Need to double check the upstream how it was handling the version there, at least I remember in some cases the app is supposed to return 404 when the `ProgramId` or `ProgramRunId` key is not found in the returned Map -- and with this change it's actually in the Map but with the actual latest version instead of the `-SNAPSHOT` version. It can lead to bunch of bugs that are hard to catch...